design/nfs/nfs-quota.md

```diff
@@ -0,0 +1,68 @@
+# NFS Quota
+
+## Background
+
+Currently, when the user creates PersistentVolumes from an NFS Rook share/export via PersistentVolumeClaim, the provisioner does not provide the spesific capacity as requested. For example the users create PersistentVolumeClaim as following:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: rook-nfs-pv-claim
+spec:
+  storageClassName: "rook-nfs-share"
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Mi
+```
+
+The client still can use the higher capacity than `1mi` as requested.
+
+This proposal is to add features which the Rook NFS Provisioner will provide the specific capacity as requested from `.spec.resources.requests.storage` field in PersistentVolumeClaim.
+
+## Implementation
+
+The implementation will be use `Project Quota` on xfs filesystem. When the users need to use the quota feature they should use xfs filesystem with `prjquota/pquota` mount options for underlying volume. Users can specify filesystem type and mount options through StorageClass that will be used for underlying volume. For example:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard-xfs
+parameters:
+  fsType: xfs
+mountOptions:
+  - prjquota
+...
+```
+
+Then the underlying PersistentVolumeClaim should be used that StorageClass 
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: nfs-default-claim
+spec:
+  storageClassName: "standard-xfs"
+  accessModes:
+  - ReadWriteOnce
+...
+```
+
+If the above conditions are met then the Rook NFS Provisioner will create projects and set the quota limit using [xfs_quota](https://linux.die.net/man/8/xfs_quota) before creating PersistentVolumes based on `.spec.resources.requests.storage` field in PersistentVolumeClaim. Otherwise the Rook NFS Provisioner will provision a PersistentVolumes without creating setting the quota.
+
+### Why XFS
```

Comment:
I haven't tested yet, but it should work as well with condition `fstype` is xfs and with `prjquota` mount options in rbd storageclass.

https://github.com/rook/rook/blob/24b0bdd119cf5cf453865af91bc81fe0ac88f5ff/cluster/examples/kubernetes/ceph/csi/rbd/storageclass.yaml#L53

Reply:
This is good to know, that Ceph underlying volumes can support this as well.  It may be *nice* to include this in the instructions, but it is definitely not a requirement to do so :)